client/controllers/events/eventIncidentReports.coffee

```diff
@@ -48,6 +49,17 @@ Template.eventIncidentReports.onCreated ->
     <% } %>
   """
 
+  @autorun =>
+    # When user modifies query, set rendering to true which triggers a loading
```

Comment:
It might simplify things to handle the feedback for filter changes in the filter or event template.